File: Assets/ShootingRPGKit/Scripts/PlayerAndCamera/UpperBodyAnimationMix.js
#pragma strict
private var mainModel : GameObject;
var upperBody : Transform;
var animationFile : AnimationClip[] = new AnimationClip[1];

function Start () {
		//For Legacy Animation.
		if(!mainModel){
			mainModel = GetComponent(Status).mainModel;
		}
		var c : int = 0;
		if(animationFile.Length > 0){
			while(c < animationFile.Length && animationFile[c]){
				mainModel.GetComponent.<Animation>()[animationFile[c].name].AddMixingTransform(upperBody);
				c++;
			}
		}
}

@script AddComponentMenu ("Shooting RPG Kit/Upper Body Animation Mix(Legacy)")

File: Assets/ShootingRPGKit/CsharpExample/ScriptCS/PlayerAndCamera/UpperBodyAnimationMix.cs
using UnityEngine;
using System.Collections;

[AddComponentMenu("Shooting-RPG Kit(CS)/Upper Body Animation Mix(Legacy)")]
public class UpperBodyAnimationMix : MonoBehaviour {
	
	private GameObject mainModel;
	public Transform upperBody;
	public AnimationClip[] animationFile = new AnimationClip[1];
	
	void Start(){
		//For Legacy Animation.
		if(!mainModel){
			mainModel = GetComponent<Status>().mainModel;
		}
		int c = 0;
		if(animationFile.Length > 0){
			while(c < animationFile.Length && animationFile[c]){
				mainModel.GetComponent<Animation>()[animationFile[c].name].AddMixingTransform(upperBody);
				c++;
			}
		}
	}
			
}
